Sources/carton/Toolchain/ToolchainManagement.swift

```diff
@@ -282,4 +282,18 @@ extension FileSystem {
 
     return AbsolutePath(binPath)
   }
+  
+  func fetchAllSwiftVersion() throws -> [String] {
+    var versions: [String] = []
+    // sdks installed by carton
+    let cartonSdkPath = homeDirectory.appending(components: ".carton", "sdk")
+    let cartonSdks = try getDirectoryContents(cartonSdkPath)
+    versions.append(contentsOf: cartonSdks)
+    // sdks installed by swiftenv
+    let swiftenvVersionPath = homeDirectory.appending(components: ".swiftenv", "versions")
+    let
+      swiftenvVersions = try getDirectoryContents(swiftenvVersionPath)
+    versions.append(contentsOf: swiftenvVersions)
+    return versions
```

Comment:
After you've separated out both of those path into computed properties, the function would fit on a single linee 🙂

```swift
func fetchAllSwiftVersions() throws -> [String] {
  try getDirectoryContents(cartonSDKPath) + getDirectoryContents(swiftenvVersionPath)
}
```